http/vertx-web-validation/src/main/java/io/quarkus/ts/vertx/web/validation/ValidationHandlerOnRoutes.java

```diff
@@ -0,0 +1,159 @@
+package io.quarkus.ts.vertx.web.validation;
+
+import static io.vertx.ext.web.validation.builder.Parameters.param;
+import static io.vertx.json.schema.common.dsl.Schemas.arraySchema;
+import static io.vertx.json.schema.common.dsl.Schemas.numberSchema;
+import static io.vertx.json.schema.common.dsl.Schemas.objectSchema;
+import static io.vertx.json.schema.common.dsl.Schemas.stringSchema;
+import static io.vertx.json.schema.draft7.dsl.Keywords.maximum;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.handler.BodyHandler;
+import io.vertx.ext.web.validation.BadRequestException;
+import io.vertx.ext.web.validation.BodyProcessorException;
+import io.vertx.ext.web.validation.ParameterProcessorException;
+import io.vertx.ext.web.validation.RequestParameters;
+import io.vertx.ext.web.validation.RequestPredicate;
+import io.vertx.ext.web.validation.RequestPredicateException;
+import io.vertx.ext.web.validation.ValidationHandler;
+import io.vertx.ext.web.validation.builder.Bodies;
+import io.vertx.ext.web.validation.builder.Parameters;
+import io.vertx.ext.web.validation.builder.ValidationHandlerBuilder;
+import io.vertx.json.schema.SchemaParser;
+import io.vertx.json.schema.SchemaRouter;
+import io.vertx.json.schema.SchemaRouterOptions;
+import io.vertx.json.schema.common.dsl.ObjectSchemaBuilder;
+
+@ApplicationScoped
+public class ValidationHandlerOnRoutes {
+    //TODO when Quarkus use vert.x version 4.4.6 we can use SchemaRepository instead of SchemaParser with SchemaRouter
+    //private SchemaRepository schemaRepository =SchemaRepository.create(new JsonSchemaOptions().setDraft(Draft.DRAFT7).setBaseUri(BASEURI));
+    private SchemaParser schemaParser;
+    private SchemaRouter schemaRouter;
+
+    @Inject
+    Vertx vertx;
+
+    private Router router;
+
+    @Inject
+    ShopResource shopResource;
+
+    private static final String ERROR_MESSAGE = "{\"error\": \"%s\"}";
+    private static final String SHOPPINGLIST_NOT_FOUND = "Shopping list not found in the list or does not exist with that name or price";
+
+    @PostConstruct
+    void initialize() {
+        router = Router.router(vertx);
+        router.route().handler(BodyHandler.create());
+        schemaParser = createSchema();
+        validateHandlerSoppingList(router);
```

Comment:
Can you please explain the intention of this `initialize` method? Honestly, I'm yet to try this locally, but to me it seems you are doing totally same thing twice - once on router you create manually and second time on router provided by Quarkus.

I believe that `VertxWebValidationIT` HTTP requests are only testing router provided by Quarkus, am I wrong? Which tests actually test your router?

Reply:
You are right, VertxWebValidationIT is testing the  router provided by Quarkus app and is not needed the specific router because with the` @Observes Router router` is enough and the key in the  `validateHandlerSoppingList` method.
Creating a new commit...to remove the other specific route.

Reply:
I will let you decide, IMO you need vertx so you could do it in observed method, but your version is also fine, not important